### PR TITLE
MDCT-2235: MLR Page 2 Filled State 

### PR DESCRIPTION
**Pull request by karla-vm**
### Description

This creates the filled state and adds a status for MLR reports.

Some notes:
- Currently, the status will stay at the default (not started) and present a warning
- The `Enter MLR` button has no functionality (yet, out of scope for this ticket)

New functionalities included here:
- Editing an MLR program
- Deleting an MLR program

### Related ticket

MDCT-2235 and MDCT-2223

### How to test

- Log in as a state user
- Create a new MLR report
- Navigate to `MLR Reporting`
- Add new program reporting information rows
- Edit that program
- Delete the program

Make sure that the MLR reports counter is going up correctly as you add new ones and that it goes down as you delete 'em.

### Important updates

N/A

### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [ ] I have updated the documentation, if necessary

### CODE REVIEW

**Review by britt-mo — COMMENTED** [review]
Should the count text  `MLR reports total: ` always be plural?

**Review by oweingrod-coforma — COMMENTED** [review]
LGTM! I think it might need to be rebased on `main` though

**Comment by karla-vm** [comment]
> LGTM! I think it might need to be rebased on `main` though

I pulled in changes from `main` do you see any conflicts I might have missed?

**Comment by karla-vm** [comment]
> Should the count text `MLR reports total: ` always be plural?

Megan gave me some direction on this, changed it in this PR: https://github.com/Enterprise-CMCS/macpro-mdct-mcr/pull/11173

File: services/ui-src/src/components/reports/ModalOverlayReportPage.tsx [hunk 1]
```diff
@@ -41,20 +48,54 @@ export const ModalOverlayReportPage = ({ route }: Props) => {
...
       )}
       <Box sx={sx.dashboardBox}>
         <Heading as="h3" sx={sx.dashboardTitle}>
           {dashTitle}
         </Heading>
-        {reportFieldDataEntities.length === 0 && (
+        {reportFieldDataEntities.length === 0 ? (
           <Box>{verbiage.emptyDashboardText}</Box>
+        ) : (
+          <>
```
Comment (ntsummers1) [inline services/ui-src/src/components/reports/ModalOverlayReportPage.tsx:83]: I think we can get rid of this tag here

File: services/ui-src/src/components/reports/ModalOverlayReportPage.tsx [hunk 2]
```diff
@@ -93,7 +145,17 @@ const sx = {
     color: "palette.gray_medium",
     textAlign: "left",
   },
-
+  header: {
+    textTransform: "none",
+    fontSize: "14px",
+    color: "#71767A",
```
Comment (ntsummers1) [inline services/ui-src/src/components/reports/ModalOverlayReportPage.tsx:151]:
```suggestion
    color: "palette.gray_medium",
```

File: services/ui-src/src/components/reports/ModalOverlayReportPage.tsx [hunk 3]
```diff
@@ -93,7 +145,17 @@ const sx = {
     color: "palette.gray_medium",
     textAlign: "left",
   },
-
+  header: {
+    textTransform: "none",
+    fontSize: "14px",
```
Comment (ntsummers1) [inline services/ui-src/src/components/reports/ModalOverlayReportPage.tsx:150]:
```suggestion
    fontSize: "sm",
```

File: services/ui-src/src/components/tables/EntityRow.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  openDeleteEntityModal,
+}: Props) => {
+  const { programName, planName } = entity;
+
+  const reportingPeriod = `${entity.reportingPeriodStartDate} to ${entity.reportingPeriodEndDate}`;
+  const eligibilityGroup = () => {
+    if (entity["eligibilityGroup-otherText"]) {
+      return entity["eligibilityGroup-otherText"];
+    }
+    return entity.eligibilityGroup[0].value;
```
Comment (ntsummers1) [inline services/ui-src/src/components/tables/EntityRow.tsx:22]: Can this ever not have a 0 index?

File: services/ui-src/src/components/tables/EntityRow.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            variant="none"
+            sx={sx.editButton}
+            onClick={() => openAddEditEntityModal(entity)}
+          >
+            {verbiage.editEntityButtonText}
+          </Button>
+        )}
+      </Td>
+      <Td>
+        {/* TODO: Enter MLR report routing */}
```
Comment (ntsummers1) [inline services/ui-src/src/components/tables/EntityRow.tsx:56]: Do we still need this TODO?

File: services/ui-src/src/components/reports/ModalOverlayReportPage.tsx (hunk 1 above)
Comment (tblackwe) [reply services/ui-src/src/components/reports/ModalOverlayReportPage.tsx:83]: agreed, do not need the wrapper `<></>` tag

File: services/ui-src/src/components/tables/EntityRow.tsx (hunk 5 above)
Comment (oweingrod-coforma) [reply services/ui-src/src/components/tables/EntityRow.tsx:56]: I think it's reasonable to leave it—this button needs logic to navigate to the MLR report overlay

**Comment by oweingrod-coforma** [comment]
> > LGTM! I think it might need to be rebased on `main` though
> 
> I pulled in changes from `main` do you see any conflicts I might have missed?

Changes look fine but it feels like there are too many commits?

File: services/ui-src/src/components/reports/ModalOverlayReportPage.test.tsx [hunk 6]
```diff
@@ -74,6 +97,20 @@ describe("Test ModalOverlayReportPage (adding new program reporting information)
     await userEvent.click(closeButton);
     expect(screen.getByTestId("modal-overlay-report-page")).toBeVisible();
   });
+
+  it("State user should be able to delete existing entities", async () => {
+    // delete program entity
+    const closeButton = screen.getByAltText("delete icon");
```
Comment (tblackwe) [inline services/ui-src/src/components/reports/ModalOverlayReportPage.test.tsx:103]: nit: we can use the getByRole to ensure element type. 
`const closeButton = screen.getByRole("button", { name: "delete icon" });`

File: services/ui-src/src/components/reports/ModalOverlayReportPage.test.tsx [hunk 7]
```diff
@@ -74,6 +97,20 @@ describe("Test ModalOverlayReportPage (adding new program reporting information)
...
+    await userEvent.click(closeButton);
+    expect(screen.getByRole("dialog")).toBeVisible();
+
+    // click delete in modal
+    const deleteButton = screen.getByText(deleteModalConfirmButtonText);
```
Comment (tblackwe) [inline services/ui-src/src/components/reports/ModalOverlayReportPage.test.tsx:108]: same nit: 
`const deleteButton = screen.getByRole("button", { name: deleteModalConfirmButtonText });`

File: services/ui-src/src/components/reports/ModalOverlayReportPage.tsx [hunk 8]
```diff
@@ -41,20 +48,52 @@ export const ModalOverlayReportPage = ({ route }: Props) => {
...
+    deleteEntityModalOnCloseHandler();
+  };
+
   return (
     <Box data-testid="modal-overlay-report-page">
       {verbiage.intro && (
         <ReportPageIntro
           text={verbiage.intro}
-          accordion={accordionVerbiage.formIntro}
+          accordion={accordionVerbiage.MLR.formIntro}
```
Comment (tblackwe) [inline services/ui-src/src/components/reports/ModalOverlayReportPage.tsx:73]: req: I'd prefer we didn't hardcode this to MLR. can we make use of the report type to determine which formIntro we should use?

File: services/ui-src/src/components/tables/EntityRow.tsx (hunk 4 above)
Comment (tblackwe) [reply services/ui-src/src/components/tables/EntityRow.tsx:22]: why is it always the first index if it isn't other-Text?

Comment (tblackwe) [reply services/ui-src/src/components/tables/EntityRow.tsx:22]: just trying to understand this code and I'm failing, no notes on how to change it

**Review by ntsummers1 — CHANGES_REQUESTED** [review]
This guy should say instructions instead of introduction

Screenshot 2023-03-27 at 11 31 38 AM

-------

1. There should be a border between the "MLR report total count" and the "Add the program..." text.
2. Also it should be MLR report total count: and not MLR report total
3. Also, the "Add the program" text is different here than it is in the Figma. Am I [looking at the wrong figma ](https://www.figma.com/file/GBktI8ZIJM9YYcdNva0ltg/MDCT---MCR?node-id=7211-151000&t=iOwpeCPJw4SqOqJA-0)(Very possible lol) 

Screenshot 2023-03-27 at 11 32 42 AM

-------
Few things on the table:

1. The error icon should be swapped to the brighter red version
2. The MCO/PIHP/PAHP name should be bolded
3. The buttons look they should be all in 1 column just with some slight padding between them? Might alleviate all the empty table headers that were introduced.

Screenshot 2023-03-27 at 11 36 51 AM

-------

This text is using color-success-darkest and should instead be using color-success-darker

Screenshot 2023-03-27 at 11 40 11 AM

-------

Mobile view is broken on this page. Now I'm hoping the statusing table isn't broken as well 😅 

https://user-images.githubusercontent.com/19439679/227992142-56edd842-a4ad-422a-bf37-33da43ee4039.mov


-----

Theres actually quite a few more small styling issues here with the table headers and table rows needing color tweeking, some padding issues, etc. Would def recommend double checking things. Hope this helps!

**Comment by karla-vm** [comment]
https://github.com/Enterprise-CMCS/macpro-mdct-mcr/pull/11170#discussion_r1149425872

@tblackwe I'm referencing the first index because it's a single radio button, but the value is contained as an object in an array. I'm not sure if there's a different way to pull in that value.

**Review by oweingrod-coforma — DISMISSED** [review]

**Comment by ntsummers1** [comment]
> @ntsummers1 I added a mobile version view of this table, could you check it out? 🙏🏽

Of course! Thanks for all the styling update. Its looking dope! 

I took a peek and noticed the widths were a little tight on the mobile view, so I quickly went in and gave them the full width when in mobile view. I've attached a quick video so you can see the difference and where I did it. Hopefully that helps? 

https://user-images.githubusercontent.com/19439679/228371671-6e534b54-cff5-4596-98fb-fc12e8345d80.mov

I think those buttons are little too cluttered though at the bottom and maybe I'm looking too deeply into things but I wonder if either you, Megan or Kim have the same thoughts about it as well? Not sure what I'd do to fix it though.. Maybe make it vertical? 🤷 Also possible its fine like that.

**Review by ntsummers1 — APPROVED** [review]

**Review by oweingrod-coforma — APPROVED** [review]

**Review by oweingrod-coforma — APPROVED** [review]

**Review by ntsummers1 — APPROVED** [review]

**Pull request merged**